awx/main/models/unified_jobs.py

```diff
@@ -376,6 +394,8 @@ def create_unified_job(self, **kwargs):
             for fd, val in eager_fields.items():
                 setattr(unified_job, fd, val)
 
+        unified_job.execution_environment = self.resolve_execution_environment()
```

Comment:
this seems fine to me, but with this addition, "execution_environment" should be removed from the list here.

https://github.com/ansible/awx/blob/4f1eafec26487892c58bbe49856c3e58755613d4/awx/main/models/jobs.py#L287

Reply:
No, maybe I'm wrong, because you check

```python
        if self.execution_environment is not None:
            return self.execution_environment
```

so I guess the pattern is that `execution_environment` on the job can be set by the unified job template, and if it is, that comes highest precedence, so it is respected.

If going _this_ route, then it seems `execution_environment` should be added to the unified job fields for inventory sources, since those have historically allowed overriding at the UJT level too.

Reply:
> this seems fine to me, but with this addition, "execution_environment" should be removed from the list here.
> 
> https://github.com/ansible/awx/blob/4f1eafec26487892c58bbe49856c3e58755613d4/awx/main/models/jobs.py#L287

What's the reasoning behind that?

Reply:
> No, maybe I'm wrong, because you check
> 
> ```python
>         if self.execution_environment is not None:
>             return self.execution_environment
> ```
> 
> so I guess the pattern is that `execution_environment` on the job can be set by the unified job template, and if it is, that comes highest precedence, so it is respected.
> 
> If going _this_ route, then it seems `execution_environment` should be added to the unified job fields for inventory sources, since those have historically allowed overriding at the UJT level too.

The `execution_environment` field on a job (as opposed to a job template) is supposed to be the already precedence-resolved EE to be used while executing.

As I noted in another comment, `InventorySource` does have `execution_environment` defined on itself, by virtue of inheriting from `UnifiedJobTemplate`.

Reply:
This is the list of fields for InventorySource to copy over when `create_unified_jobs` is called

https://github.com/ansible/awx/blob/4f1eafec26487892c58bbe49856c3e58755613d4/awx/main/models/inventory.py#L1030-L1034

This should contain `execution_environment` by the same measure that JT does, if that's the design.

Reply:
It made sense when I did it in the first PR, but I suspect that you are right that it is no longer needed then.